gcp_query_tools/connect_analytics_helper_functions.R

```diff
@@ -245,52 +246,48 @@ add_novel_vars_to_csv <- function(vars, in_file, out_file = NULL) {
   #return(vars_to_export)
 }
 
-get_list_of_unique_responses <- function(var_name, project, table){
-  #rebecca test comment
-  # Description:
-  #   This function queries a variable (that returns a json array in GCP) 
-  #   specified by var_name within a given project and table, and generates a 
-  #   list of unique values or responses that appear in the json arrays for that 
-  #   variable.
-  # Inputs:
-  #   var_name: The name of the variable of interest as a string.
-  #   project:  The GCP project ID as a string.
-  #   table:    The name of the table as a string (e.g., "Database.table_name")
-  # Returns:
-  #   unique_responses: A character array of the unique values or responses 
-  #                     returned by the variable across all rows of the table.
-  # TODO: Take an array of variables as an input and write one query for all 
-  #       of them. There is a 10MB minimum charge, so combining running one 
-  #       query for all of the variables would be more cost efficient.
-  
-  # Query data for this variable
-  tab_path <- paste0("`", project, ".", table, "`") # Format table path for SQL
-  bq_query <- paste("SELECT", var_name, "AS var FROM", tab_path,  
-                    "WHERE",  var_name, "IS NOT NULL", sep=" ")
-  bq_table <- bq_project_query(project, bq_query) # 
-  df_var   <- bq_table_download(bq_table, bigint = "integer64")
-  
-  # Build list of unique responses for this variable
-  unique_responses <- c()
-  for (i in 1:nrow(df_var)){
-    responses           <- df_var$var[[i]]
-    responses_available <- length(responses) > 0
-    if (responses_available){
-      for (j in 1:length(responses)){
-        response          <- responses[j]
-        response_is_novel <- !(response %in% unique_responses)
-        if (response_is_novel){
-          unique_responses <- append(unique_responses, response)
-        }
-      }
+
+get_unique_values <- function(project, table, array_vars) {
+  
+  # Initialize a string to store the SQL query
+  sql_query <- ""
+  
+  # Loop through each array variable to generate the SQL query
+  for (array_var in array_vars) {
+    
+    # Add to the SQL query a SELECT DISTINCT statement to get unique values for the array variable
+    sql_query <- glue("{sql_query}
+      SELECT DISTINCT
+        '{array_var}' AS variable,  # Label the array variable
```

Comment:
This is an interesting way to construct the query in a loop. I'm not sure I understand it to be honest, but I know you've tested it. I might have you explain this to me sometime!